src/docs/binding-typedatabase.md

```diff
@@ -0,0 +1,58 @@
+# Binding Type Database
+
+In binding a Swift module, there will be nominal type references presented to the binder that have been bound previously. The binder will need to perform a few important operations on these types:
+
+1. Determine the C# type that should be used in the public-facing API and what namespaces need to be references
+2. Determine the size, stride, and alignment of value types that contain the type reference, if known at binding time.
+
+After binding the module, the binder will have new type database entries to write.
+
+Therefore, it makes sense to aggregate a collection of type database files for each module. The binding tool can either read them all in at the start of the world or read them in lazily for each module reference.
+
+The file format for the type database files is not vitally important beyond the fact that it should be easily serializable. Either XML or JSON would be fine.
+
+Here are the elements that are likely to be needed in a type database entry:
+
+- Swift module name
+- Swift type name (may be a type path for inner types)
+- Size, stride, and alignment (if knowable)
+- C# namespace
+- C# type name (may be a type path for inner types)
+- Swift entity type (stuct, enum, class, actor, protocol)
```

Comment:
I think we will need information whether the type is frozen or not, as this will change the way we bind references to this type (e.g. when binding a function whose signature contains a non-frozen struct we will need to replace the struct argument with pointer argument)